unstructured_ingest/v2/processes/connectors/astradb.py

```diff
@@ -45,6 +89,154 @@ class AstraDBUploadStagerConfig(UploadStagerConfig):
+    def astra_record_to_file_data(self, astra_record: Dict[str, Any]) -> FileData:
+        return FileData(
+            identifier=astra_record["_id"],
+            connector_type=CONNECTOR_TYPE,
+            source_identifiers=SourceIdentifiers(
+                fullpath=astra_record["_id"] + ".txt",
+                filename=astra_record["_id"] + ".txt",
+                rel_path=astra_record["_id"] + ".txt",
+            ),
+            additional_metadata=astra_record.get("metadata", {}),
+        )
+
+    @requires_dependencies(["astrapy"], extras="astradb")
+    def run(self, **kwargs: Any) -> Generator[FileData, None, None]:
+        # Get the collection
+        collection = self.get_collection()
+
+        # Perform the find operation to get all items
+        astra_db_docs_cursor = collection.find({}, projection={"_id": True, "metadata": True})
+
+        # Iterate over the cursor
+        astra_db_docs = []
+        for result in astra_db_docs_cursor:
+            astra_db_docs.append(result)
+
+        # Create file data for each astra record
+        for astra_record in astra_db_docs:
+            file_data = self.astra_record_to_file_data(astra_record=astra_record)
+
+            yield file_data
+
+
+@dataclass
+class AstraDBDownloader(Downloader):
+    connection_config: AstraDBConnectionConfig
+    download_config: AstraDBDownloaderConfig
+    connector_type: str = CONNECTOR_TYPE
+
+    @SourceConnectionNetworkError.wrap
+    def _fetch_record(self, file_data: FileData):
+        collection = get_astra_collection(
+            connection_config=self.connection_config,
+            config=self.download_config,
+        )
+
+        record = collection.find_one({"_id": file_data.identifier})
+
+        return record
+
+    def get_download_path(self, file_data: FileData) -> Path:
+        # Get the relative path from the source identifiers
+        rel_path = ""
+        if file_data.source_identifiers and file_data.source_identifiers.rel_path:
+            rel_path = file_data.source_identifiers.rel_path
+
+        # Remove leading slash if present
+        rel_path = rel_path[1:] if rel_path.startswith("/") else rel_path
+
+        return self.download_dir / Path(rel_path)
+
+    @SourceConnectionError.wrap
+    def run(self, file_data: FileData, **kwargs: Any) -> download_responses:
+        # Fetch the record from the collection
+        record = self._fetch_record(file_data=file_data)
+
+        # Get the download path
+        download_path = self.get_download_path(file_data=file_data)
+        download_path.parent.mkdir(parents=True, exist_ok=True)
+
+        # Write "record" to a json file at download path
+        with open(download_path, "w") as file:
+            # List comp. to prevent newlines at the end of the file which affects partitioning
+            file.write("\n".join([str(x) for x in record.values()]))
+
+        return DownloadResponse(file_data=file_data, path=download_path)
```

Comment:
Will want to wrap this like [this](https://github.com/Unstructured-IO/unstructured-ingest/blob/main/unstructured_ingest/v2/processes/connectors/google_drive.py#L303):
```python
return self.generate_download_response(file_data=file_data, download_path=download_path)

```